src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -96,34 +96,61 @@ private void LoadingStatusIndicator_PropertyChanged(object sender, PropertyChang
             });
         }
 
-        // Indicates wether check boxes are enabled on packages
-        private bool _checkBoxesEnabled;
+        public bool CheckBoxesEnabled { get; set; }
 
-        public bool CheckBoxesEnabled
+        public bool IsSolution { get; set; }
+
+        public ObservableCollection<object> Items { get; } = new ObservableCollection<object>();
+        private ICollectionView CollectionView
         {
             get
             {
-                return _checkBoxesEnabled;
+                return CollectionViewSource.GetDefaultView(Items);
             }
-            set
-            {
-                _checkBoxesEnabled = value;
+        }
 
-                if (!_checkBoxesEnabled)
+        /// <summary>
+        /// Count of Items (excluding Loading indicator) that are currently shown after applying any UI filtering.
+        /// </summary>
+        private int FilterCount
+        {
+            get
+            {
+                if (CollectionView.Filter != null)
                 {
-                    // the current tab is not "updates", so the container
-                    // should become invisible.
-                    _updateButtonContainer.Visibility = Visibility.Collapsed;
+                    var listCollectionView = (ListCollectionView)CollectionView;
+                    return listCollectionView.OfType<PackageItemListViewModel>().Count();
```

Comment:
Use `.Count` instead, no need to enumerate the complete list.

Reply:
I don't have a list, I have an IEnumerable. I actually looked this up before I wrote it, and .Count() on an Enumerable is twice as fast compared to .ToList().Count().
(good read) https://stackoverflow.com/a/33120649

Reply:
ListCollectionView has a .Count

https://docs.microsoft.com/en-us/dotnet/api/system.windows.data.listcollectionview.count?view=netcore-3.1#System_Windows_Data_ListCollectionView_Count. 

Does the PackageItemListViewModel type matter? 


>  Enumerable is twice as fast compared to .ToList().Count().

Oh I get that. ToList enumerates it already, Count() is an IEnumerable method that runs enumeration, so it totally makes sense. I was recommending using the List directly, which itself has a Count prop

Reply:
Based on feedback from sbanni, I've actually removed this logic. I'm accessing the IEnumerable directly and returning count (no more ListCollectionView). 

(Just to re-iterate the now irrelevant discussion: OfType provides an IEnumerable, which does not have a Count property. 
Maybe there's only ever 1 object that's not a VM.... but that's more of a class redesign in a future PR, b/c a lot of existing logic looks for that LoadingStatus object)

Reply:
> (Just to re-iterate the now irrelevant discussion: OfType provides an IEnumerable, which does not have a Count property.


Yep, I understand that, hence why I asked `Does the PackageItemListViewModel type matter?`. 

Resolving this thread now.